src/main/java/de/tum/in/www1/artemis/domain/Feedback.java

```diff
@@ -69,6 +70,27 @@
     @ManyToOne
     private GradingInstruction gradingInstruction;
 
+    /**
+     * Represents the reference of the previously assessed block, whose feedback we are reusing
+     */
+    @Transient
+    @JsonSerialize
```

Comment:
why do we need the `@JsonSerialize`? what does it do?

Reply:
This field alone is supposed to _not_ be persisted in the database, therefore the '@ Transient'. So, you just send the data back when requested. However I was never getting the data back regardless. After couple of hours of 🔍 , it appears that JPA saves the field anyway temporarily ( for subsequent requests to avoid xtra calculations ) for which it needs to be "serialized" - therefore _JSONSerialize_ is used here. I don't know where it decided to not return the field at all, but I can assume it tries to save it assuming its serialized, and when fetched back it is deserialized as garbage and then ignored.

Reply:
ah...thanks for clarification 😄 I can imagine that you spent some hours there... 🤯